dataStructures/queues/queue_test.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestNewOrder(t *testing.T) {
+	cases := []struct{
+		name, product, customerName string
+		priority, quantity int
+		want *Order
+	}{
+		{"create new order", "macbook", "andela", 1, 20, &Order{1, 20, "macbook", "andela"}},
+		{"create another order", "windows", "apple", 2, 30, &Order{2, 30, "windows", "apple"}},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ord := newOrder(tc.priority, tc.quantity, tc.product, tc.customerName)
+			if ord.customerName != tc.want.customerName {
```

Comment:
You can save a lot of code by comparing the order as a whole thing. Some options are:

```Go
// In tests it's common to call the value that you're testing "got"
// If there are lots of them then "gotOrder" "gotQueue" etc. but in this test it doesn't matter
// We do it because then someone reading the code immediately knows that this is the value being tested
got := newOrder(…)

if !reflect.DeepEqual(got, tc.want) {
  // It's important to print the name of the function being tested _and_ the input
  // arguments too. Even if the test name is TestNewOrder, an error message
  // should contain all the information needed to debug it.
  t.Errorf("newOrder(%d, %d, %q, %q) got %+v; want %+v", tc.priority, tc.quantity, tc.product, tc.customerName, got, tc.want)
}

// or

// If your struct is really big then it's better to have a diff
// Using https://godoc.org/github.com/google/go-cmp/cmp#Diff  (look at the Example on godoc)
// This makes it much easier to debug
// Note how I call Diff(want, got) not Diff(got, want)
// This is because then the diff message prints - in front of missing lines and + in front of added lines
// It's so small but so much easier for people to understand when debugging!
if diff := cmp.Diff(tc.want, got); diff != "" {
  t.Errorf("newOrder(… [like above]…) diff -want +got: %s", …, diff)
}
```